Added Category pages & closed #20

rosnovsky opened the pull request:
Added Category pages:

- added slug field to `category` schema
- added category pages to `gatsby-node.js`
- added `category.js` ~page~ template to Gatsby; **TODO**: add categories page featuring all categories
- added links to Categories in the post sidebar

rosnovsky committed:
Added Category pages & Closed #20

rosnovsky committed:
Added Category pages & Closed #20

rosnovsky committed:
Generate lockfiles

rosnovsky committed:
Added Category pages & Closed #20

rosnovsky committed:
moved `category` from pages to templates, added a check for no-slug categories

rosnovsky committed:
Generate lockfiles

rosnovsky committed:
fixed `package.json` formatting

kmelve commented on template/package.json:
```diff
@@ -1,20 +1 @@
-    "lerna": "^3.13.1"
-  }
-}
+{"private":true,"name":"blog-with-gatsby","version":"1.0.17","scripts":{"build":"lerna run build --parallel","dev":"lerna run dev --parallel","format":"lerna run format","build-studio":"lerna bootstrap && cd studio && npm run build","build-web":"lerna bootstrap && (cd studio && SANITY_AUTH_TOKEN=$SANITY_DEPLOY_STUDIO_TOKEN npm run graphql-deploy) && (cd web && npm run build)","graphql-deploy":"le ...
```
Not sure why you removed the line spaces here?

rosnovsky replied:
Stupid prettier :( I though I fixed it, but apparently I didn't. Would you suggest preserving the results of `npm run format`? It changes quite a bit of formatting :/

kmelve commented on template/web/gatsby-node.js:
```diff
@@ -11,9 +11,7 @@ async function createBlogPostPages (graphql, actions) {
-      allSanityPost(
-        filter: { slug: { current: { ne: null } }, publishedAt: { ne: null } }
-      ) {
+      allSanityPost(filter: {slug: {current: {ne: null}}, publishedAt: {ne: null}}) {
```
Might be a prettier or eslint thing(?), but I think the original formatting is more readable, might even add more lines between the different filters

rosnovsky replied:
This is an example of what running `npm run format` results in. Not sure how to instruct `prettier-eslint` to break lines in such cases. Will fix manually for now.

kmelve commented on template/web/gatsby-node.js:
```diff
@@ -46,6 +44,70 @@ async function createBlogPostPages (graphql, actions) {
+
+  if (result.errors) throw result.errors
+
+  const categoryNodes = (result.data.allSanityCategory || {}).edges || []
```
Maybe we should use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) instead?

kmelve commented on template/web/gatsby-node.js:
```diff
@@ -46,6 +44,70 @@ async function createBlogPostPages (graphql, actions) {
+
+  categoryNodes.forEach(node => {
+    const {id, slug} = node.node
+    if (!id || !slug.current) return 'No category id or slug!'
```
Prefer braces `{}` for conditional statements. Are you sure we should return anything here btw? Is it better to use the Gatsby reporter for this?

rosnovsky replied:
yep, `reporter.warn` is most suitable here (it's not an error per se if there are no categories, yet it's something to pay attention to because it's likely that you forgot to update `Category` schema with a `slug`)

kmelve reviewed (changes requested):
Some small nitpicks, but I think this is allover good! Thanks for putting in the time 🙌 

DM me a projectID of yours, and I'll set you up with our special community plan.

rosnovsky committed:
Applied changes requested by @kmelve

rosnovsky commented:
- fixed `package.json` formatting (hopefully!)
- fixed GraphQL blog post query formatting (fingers crossed!)
- updated empty categories situation with `reporter.warn` to let folks know we couldn't build category pages and then proceed
- updated query results check with optional chaining. Added `"engines": { "node": ">=14.14.0" }` to `package.json` since optional chaining is only officially made available in Node 14.6, yet I was only able to make it work with 14.14

🤞
